mono/mini/aot-compiler.c

```diff
@@ -8634,10 +8635,15 @@ add_gsharedvt_wrappers (MonoAotCompile *acfg, MonoMethodSignature *sig, gboolean
 	}
 
 #ifndef MONO_ARCH_HAVE_INTERP_ENTRY_TRAMPOLINE
-	if (interp_in) {
+	/*
+	 * Interpreter methods in llvmonly+interp mode are called using gsharedvt_in wrappers,
+	 * since we already generate those in llvmonly mode. But methods with a large
+	 * number of arguments need special processing (see interp_create_method_pointer_llvmonly),
+	 * which only interp_in wrappers do.
+	 */
+	if (interp_in || (gsharedvt_in && sig->param_count > MAX_INTERP_ENTRY_ARGS)) {
```

Comment:
Aren't we adding interp wrappers also on non interp aot images here ? While there aren't probably many methods satisfying this condition, looks rather ugly if so.

Reply:
Will fix.